# Jane Street Training set correlations
[Kaggle Jane Street Market Prediction](https://www.kaggle.com/c/jane-street-market-prediction/overview)


In [1]:
import pandas as pd
import numpy as np
import datatable as dt
X = dt.fread('jane-street-market-prediction/train.csv').to_pandas()
#X = dt.fread('jane-street-market-prediction/train.csv', max_nrows=10000, skip_to_line=520000).to_pandas()

In [2]:
X = X[(X.date > 85) & (X.date < 460)] 

In [3]:
corr = X.corr()

In [4]:
pd.set_option('display.max_rows', None)

In [5]:
corr.resp.sort_values()

feature_17    -0.026171
feature_37    -0.024880
feature_21    -0.023131
feature_23    -0.019968
feature_0     -0.017129
feature_38    -0.015160
feature_25    -0.014984
feature_18    -0.012228
feature_22    -0.010672
feature_19    -0.010612
feature_24    -0.009154
feature_103   -0.009130
feature_52    -0.008630
feature_51    -0.008050
feature_50    -0.007169
feature_106   -0.006697
feature_97    -0.006268
feature_77    -0.005677
ts_id         -0.005451
feature_41    -0.005342
weight        -0.005341
date          -0.005302
feature_83    -0.005292
feature_26    -0.005043
feature_105   -0.004634
feature_45    -0.004123
feature_49    -0.003420
feature_92    -0.003353
feature_117   -0.003167
feature_104   -0.002871
feature_116   -0.002668
feature_90    -0.002594
feature_118   -0.002414
feature_20    -0.002370
feature_115   -0.002227
feature_72    -0.002174
feature_93    -0.002015
feature_114   -0.001907
feature_100   -0.001763
feature_68    -0.001726
feature_67    -0.001468
feature_48    -0

In [1]:
import pandas as pd
import numpy as np
import datatable as dt
X = dt.fread('jane-street-market-prediction/train.csv').to_pandas()
#X = dt.fread('jane-street-market-prediction/train.csv', max_nrows=10000, skip_to_line=520000).to_pandas()

In [20]:
x27 = X.groupby('date').feature_27.median()
x27[x27.index == ( X.date+1 )]

ValueError: Lengths must match to compare

In [2]:
Xt = X[X.date >= 460]
yt = np.where((Xt.resp > 0), 1, 0)
X = X[(X.date > 85) & (X.date < 460)]
y = np.where((X.resp > 0), 1, 0)
drop_cols = ['date', 'resp_1', 'resp_2', 'resp_3', 'resp_4', 'resp', 'ts_id']
X.drop(drop_cols, axis=1, inplace=True)
Xt.drop(drop_cols, axis=1, inplace=True)

In [3]:
from sklearn.experimental import enable_hist_gradient_boosting
from sklearn.ensemble import HistGradientBoostingClassifier
clf = HistGradientBoostingClassifier(max_iter=100, verbose=0, random_state=42).fit(X, y)

In [4]:
clf.score(Xt, yt)

0.5203650466860196

### max_iter=100, verbose=0, random_state=42
0.5211998142876572
kaggle 5171.245

### drop feature_61
0.5215234039759324

### drop feature_61, 'feature_64'
0.5201024231708976

### drop feature_61, 'feature_90
0.5203650466860196